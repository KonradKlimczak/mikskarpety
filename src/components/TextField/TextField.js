import React, { Component } from "react";
import classnames from "classnames";
class TextField extends Component {
    render() {
        return (
            <div className={classnames("text-field")}>
                <div className={classnames("text-field--icon")}>
                    <img src={this.props.icon} />
                </div>
                <input placeholder={this.props.placeholder} />
            </div>
        );
    }
}

export default TextField;
