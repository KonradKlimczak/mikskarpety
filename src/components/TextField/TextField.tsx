import classnames from 'classnames';
import * as React from 'react';

interface TextFieldProps {
  icon?: string;
  placeholder: string;
  debounce?: number;
  onChange: (value: string) => void;
}

const TextField: React.SFC<TextFieldProps> = props => {
  const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
    if (props.debounce) {
      debounce(() => props.onChange(event.target.value), props.debounce);
    } else {
      props.onChange(event.target.value);
    }
  };

  return (
    <div className={classnames('text-field')}>
      <div className={classnames('text-field--icon')}>
        <img src={props.icon} />
      </div>
      <input placeholder={props.placeholder} onChange={handleChange} />
    </div>
  );
};

export default TextField;
