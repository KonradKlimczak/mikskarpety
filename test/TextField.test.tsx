import React from 'react';
import renderer from 'react-test-renderer';
import { render, fireEvent, cleanup } from 'react-testing-library';

import TextField from '../src/components/TextField/TextField';

afterEach(cleanup);

it('TextField renders correctly', () => {
  const tree = renderer.create(<TextField icon='TestIcon' placeholder='Find love' onChange={console.log} />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('TextField onChange handler', () => {
  const changeCallback = jest.fn();
  const textFieldRenderer = render(<TextField icon='TestIcon' placeholder='Find love' onChange={changeCallback} />);
  const testInput = textFieldRenderer.getByTestId('input');
  expect(testInput).toBeTruthy();
  const value = 'TEST';
  fireEvent.change(testInput, { target: { value } });
  expect(changeCallback).toBeCalledTimes(1);
  expect(changeCallback).toBeCalledWith(value);
});
