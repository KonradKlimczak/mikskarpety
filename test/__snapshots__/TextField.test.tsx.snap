// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextField renders correctly 1`] = `
<div
  className="text-field"
>
  <div
    className="text-field--icon"
  >
    <img
      src="TestIcon"
    />
  </div>
  <input
    data-testid="input"
    onChange={[Function]}
    placeholder="Find love"
  />
</div>
`;
